### ISSUE TITLE
WikiTables Semantic Parser

### ISSUE THREAD

**Issue by schmmd** [allenai/allennlp#1123]

**Comment by BladeSun** [comment]
Hi, could you provide the DPD file?

**Comment by matt-gardner** [comment]
We got it from here: http://cs.stanford.edu/~ppasupat/research/h-strict-all-matching-lfs.tar.gz.  Though, see this paper: https://www.semanticscholar.org/paper/It-was-the-training-data-pruning-too!-Mudrakarta-Taly/09b94316daac1bb88e9a052aa2f8f662c1c6c469.  And also note that our current implementation of this parser is about 4.5-5 points behind the original implementation.  We're stuck on figuring out what the cause of this difference is.  We're waiting on cleaning up the code and writing tutorials for how to use the semantic parsing stuff until we've matched the performance of the parser (and we're pretty busy with EMNLP for the next week, anyway).

**Comment by BladeSun** [comment]
@matt-gardner Thank you very much! Good luck with your EMNLP!

**Comment by dungtn** [comment]
Hi, how I can run the preprocess_wikitables.py script?

**Comment by matt-gardner** [comment]
We're not quite ready to fully support this code yet; I'll be substantially cleaning it up in the next couple of weeks, now that the EMNLP deadline has passed.  We're aiming to include the parser, a demo, and some nice tutorials in an upcoming major release of AllenNLP.  Until then, things are a bit rough.

For now, download the data from the dataset's website, download the DPD file linked above, set up the config file to point to the right files (see the example in `training_config/`, but change the data paths to point to `.examples` files instead of `.jsonl` files), and run the script.  If you need more help than that, it's probably best to wait until we have some tutorials written.

**Comment by dungtn** [comment]
Thank you for the prompt reply :-) I've just found out how to run it. However, there are lots of missing DPD (I'm trying to load the random split1), is it normal? 

I'm looking forward to the documentations. Please let me know if there is anyway I can help. All the bests and good luck with EMNLP.

**Comment by matt-gardner** [comment]
Yes, only about 2/3 of the data has DPD logical forms, which means you're throwing out a bunch of your training data right off the bat.  Our EMNLP submission tries to address this, at least in part...

**Comment by dungtn** [comment]
That sounds interesting. Too bad we have to wait until August to check out your paper :-)

**Comment by dungtn** [comment]
Hi, can you share the hyper-parameters/random seed of the model? So far I can only get 35% accuracy.

**Comment by matt-gardner** [comment]
Yes, I mentioned above that we were ~5% behind the original parser.  When I ran the original parser on the same data that I gave this, I got around 40.5% on dev.  The highest I've gotten with this implementation is ~35.7.  My current guess is that this is due to how we're handling things like `(number 1)` in the logical form, but I'm still looking into this.

**Comment by MQ-MJ** [comment]
Thanks for making this available.  We are thinking of starting a project on semantic parsing on Wikitables, and I'm wondering whether to use this implementation or the original Java/Scala implementation as the starting point.  What do you recommend?  Are you still 5% behind the original parser?  That's a big performance gap.

Also, any chance of getting a preprint of your EMNLP submission?  (smile)

Thanks again for this wonderful work,

Mark

**Comment by matt-gardner** [comment]
The wikitables parser is merged into master, the demo will be live on demo.allennlp.org soon, and we're not doing any more development on the SEMPRE lambda-DCS version.  I'm guessing that the remaining ~2% discrepancy is due to differences in lemmatization and number detection between spacy and CoreNLP - we found some errors where, e.g., "Russian" didn't match to "Russia" because they didn't map to the same lemma, and other similar issues.

We _are_ still working on the simpler language in the MAPO paper, and I'm hopeful that that will give improved performance.  But, I'm closing this issue as finished.  Official launch of the semantic parsing framework should happen within the next two weeks.

**Comment by gagankhanijau** [comment]
Hi @matt-gardner, the link you have provided above for the dpd denotations doesn't exist anymore. 
I found some resources from  P. Pasupat's GitHub but the h-strict-all-matching-lfs.tar.gz doesn't exist anymore.  https://nlp.stanford.edu/software/sempre/wikitable/dpd/

would the whole dump of denotations work? its the first zip on the above link.

**Comment by matt-gardner** [comment]
The last zip file is the file that we used.  Note, though, that this is using additional human annotation than the first file.

**Comment by gagankhanijau** [comment]
Hi @matt-gardner thanks for a quick reply. The last zip file doesn't exist anymore on this link as well..is there any other way to get that zip ?

**Comment by YoPatapon** [comment]
I found the new download link [here](https://nlp.stanford.edu/software/sempre/wikitable/dpd/).

**Comment by gagankhanijau** [comment]
@YoPatapon I have checked that link but the fourth, filtered logical forms are not there if you try to download.

**Comment by YoPatapon** [comment]
Yes, you're right. I only downloaded the first, all logical forms.

**Comment by YoPatapon** [comment]
@matt-gardner Hi, Matt. I noticed that you mentioned your original implementation achieves 40.5% accuracy on dev set. But the accuracy reported in the paper is 42.7% with a single model. Is it a average accuracy of k-fold validation or what? Thanks.

**Comment by matt-gardner** [comment]
The ~40.5 number for the original implementation was calculated on the dev set of split 1, training with 10 logical forms.  This is one run of one fold of what's in [table 4](https://www.semanticscholar.org/paper/Neural-Semantic-Parsing-with-Type-Constraints-for-Krishnamurthy-Dasigi/8c6f58ed0ebf379858c0bbe02c53ee51b3eb398a/figure/5) in the original paper.  That's the setting we were starting with when re-implementing it, so that's the number we were comparing against.
